corebuild/integration/ILLink.Tasks/LinkTask.cs

```diff
@@ -12,16 +12,42 @@ namespace ILLink.Tasks
 	public class ILLink : Task
 	{
 		/// <summary>
-		///   Paths to the assembly files that should be considered as
-		///   input to the linker. Currently the linker will
-		///   additionally be able to resolve any assemblies in the
-		///   same directory as an assembly in AssemblyPaths, but this
-		///   behavior should not be relied upon. Instead, work under
-		///   the assumption that only the AssemblyPaths given will be
-		///   resolved.
+		///   Paths to the application assembly files that should be considered as
+		///   input to the linker.
 		/// </summary>
 		[Required]
-		public ITaskItem[] AssemblyPaths { get; set; }
+		public ITaskItem [] ApplicationAssemblyPaths { get; set; }
+
+		/// <summary>
+		///   Paths to the .NET Core Framework assembly files that should be considered as
+		///   input to the linker.
+		/// </summary>
+		[Required]
+		public ITaskItem  [] NetCoreFrameworkAssemblyPaths { get; set; }
+
+		/// <summary>
+		///   Action to perform on used application assemblies.
+		/// </summary>
+		[Required]
+		public ITaskItem UsedApplicationAssemblyAction { get; set; }
+
+		/// <summary>
+		///   Action to perform on unused application assemblies.
+		/// </summary>
+		[Required]
+		public ITaskItem UnusedApplicationAssemblyAction { get; set; }
+
+		/// <summary>
+		///   Action to perform on used .NET Core Framework assemblies.
+		/// </summary>
+		[Required]
+		public ITaskItem UsedNetCoreFrameworkAssemblyAction { get; set; }
+
+		/// <summary>
+		///   Action to perform on unused .NET Core Framework assemblies.
+		/// </summary>
+		[Required]
+		public ITaskItem UnusedNetCoreFrameworkAssemblyAction { get; set; }
 
```

Comment:
In order to keep the ILLink task easy to use, I don't think we should introduce separate inputs for used/unused app/framework assemblies. The idea is that someone familiar with the mono linker could easily use LinkTask without the rest of the publish targets to implement a custom linker workflow.

I'd suggest computing the action per assembly outside of this msbuild task, somewhere in the publish logic in ILLink.Tasks.targets. Then I can think of two good ways to pass the actions through to this task:
- add an AssemblyAction metadata to the input AssemblyPaths
- introduce an input itemgroup AssemblyActions that contains assembly names with Action metadata

The first feels cleaner and less redundant to me, so that would be my suggestion.